src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -967,7 +967,11 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                 } else if (callback instanceof ServerCredentialCallback) {
                     final ServerCredentialCallback serverCredentialCallback = (ServerCredentialCallback) callback;
 
-                    CredentialSource serverCredentialSource = stateRef.get().getMechanismConfiguration().getServerCredentialSource();
+                    MechanismConfiguration mechnismConfiguration  = stateRef.get().getMechanismConfiguration();
+                    if (mechnismConfiguration == null) {
+                        throw log.emptyMechnismConfiguration();
+                    }
+                    CredentialSource serverCredentialSource = mechnismConfiguration.getServerCredentialSource();
```

Comment:
please fix this type, which you repeat on most of lines of your commit: "mechnism" -> "mechanism"